src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectWriter.cs

```diff
@@ -231,6 +231,7 @@ private void Write(WriteObjectInfo objectInfo, NameInfo memberNameInfo, NameInfo
                 }
             }
 
+            Debug.Assert(memberNameInfo != null); // TODO this is strange, doing null check in above and below statemments and not checking here
             if (memberNameInfo._isParentTypeOnObject)
```

Comment:
This is strange that in above and below statements first null check memberNameInfo but not here